jsonrpc2.go

```diff
@@ -366,11 +366,10 @@ type Conn struct {
 
 	h Handler
 
-	mu       sync.Mutex
-	shutdown bool
-	closing  bool
```

Comment:
Since there is now a single closing procedure, there doesn't appear to be the need for a `closing` AND `shutdown` state any more.